lms/djangoapps/discussion/views.py

```diff
@@ -462,9 +467,20 @@ def user_profile(request, course_key, user_id):
                 'profiled_user': profiled_user.to_dict(),
                 'threads': threads,
                 'user_info': user_info,
+                'roles': utils.get_role_ids(course_key),
+                'can_create_comment': has_permission(request.user, "create_comment", course.id),
+                'can_create_subcomment': has_permission(request.user, "create_sub_comment", course.id),
+                'can_create_thread': has_permission(request.user, "create_thread", course.id),
+                'flag_moderator': bool(
+                    has_permission(request.user, 'openclose_thread', course.id) or
```

Comment:
I agree that this cast seems unnecessary, as I can't see a way in which this would not always be a boolean already.